### PR TITLE
Ensure failoverCandidates have no servers not in Plan.

### PR DESCRIPTION
**Pull request by neunhoef**
Fix failoverCandidates for servers not in Plan.

Problem: we must never fail over to a server which is not in Plan
because it might not have the data.

ToDo:

 - [X] AddFollower, FailedFollower, CleanOutServer and MoveShard will not
       add a server to the Plan which is in failoverCandidates for this
       shard or any of the ones with distributeShardsLike.
 - [X] add preconditions to ensure that it is snapshot-safe
 - [X] make dbserver Maintenance remove servers that are not in Plan from 
       failoverCandidates.

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12183/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12189/ new try after fixing compilation.

File: arangod/Cluster/Maintenance.cpp [hunk 1]
```diff
@@ -233,19 +233,32 @@ void handlePlanShard(uint64_t planIndex, VPackSlice const& cprops, VPackSlice co
...
+        if (shards.isObject()) {
+          VPackSlice planServers = shards.get(shname);
+          if (planServers.isArray()) {
             std::unordered_set<std::string> followersToDrop;
-            for (auto const& q : VPackArrayIterator(inSyncFollowers)) {
-              followersToDrop.insert(q.copyString());
+            // Now we have two server lists (servers and
+            // failoverCandidates, we are looking for a server which
+            // occurs in either of them but not in the plan
+            VPackSlice serverList = lcol.get("servers");
```
Comment (mchacki) [inline arangod/Cluster/Maintenance.cpp:248]:
```suggestion
            VPackSlice serverList = lcol.get(SERVERS);
```

**Review by mchacki — CHANGES_REQUESTED** [review]
Code LGTM.

But we are loosing one static string here, see comment

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12205/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12205/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12212/

**Comment by KVS85** [comment]
Tests blue.

**Review by KVS85 — APPROVED** [review]

**Pull request merged**